x-pack/elastic-agent/pkg/fleetapi/action.go

```diff
@@ -83,6 +92,33 @@ func (a *ActionConfigChange) ID() string {
 	return a.ActionID
 }
 
+// ActionUpgrade is a request for agent to upgrade.
+type ActionUpgrade struct {
+	ActionID   string `json:"id" yaml:"id"`
+	ActionType string `json:"type" yaml:"type"`
+	Version    string `json:"version" yaml:"version"`
+	SourceURI  string `json:"source_uri" yaml:"source_uri"`
```

Comment:
fleet can tell us where binary is located, maybe it is our official store but in case it is not, it can be kibana api or some fileserver within a company

Reply:
I would keep it simple for now and make the upgrade with and without fleet identical. The Agent knows where to pull data from. We can extend it later on if we actually support different sources. My hope is that this makes the API simpler as Fleet only needs to send down the version and does not need to also figure out how the url looks, Agent already has this built in.

Reply:
@ruflin I don't think we should remove this, in isolated environments in most cases I would believe that a mirror of the builds will be provided at a different sourceURI.

Reply:
@blakerouse But can we make it optional for now to not need to build this knowledge also into Kibana?

Reply:
Yes I think it should be optional, if the field is not provided then Agent fallsback to the default artifacts API.